P202-Training.Domain/IMappingService.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace P202.Training.Domain
+{
+    public interface IMappingService
+    {
+        TDest Map<TSrc, TDest>(TSrc source) where TDest : class;
```

Comment:
I think we need to enforce that `TSrc` and `TDest` cannot be the same class. Otherwise you might end up having service methods that receive a class to the domain and map it to the same class, which is what happened in the service classes in here. Also the tests have that issue. We need to fix that.

I'd have a base class for DTOs and a base class for Domain entities. That way you can define two methods. One for mapping a DTO to a Domain entity, and viceversa.

Reply:
In this point, I gonna need help

Reply:
No problem. Let's discuss after the meeting.